src/function/scalar/date/date_trunc.cpp

```diff
@@ -11,216 +11,303 @@
-		return MillenniumTruncOperator::Operation<TA, TR>(element);
+		return DateTrunc::MillenniumOperator::Operation<TA, TR>(element);
 	case DatePartSpecifier::CENTURY:
-		return CenturyTruncOperator::Operation<TA, TR>(element);
+		return DateTrunc::CenturyOperator::Operation<TA, TR>(element);
 	case DatePartSpecifier::DECADE:
-		return DecadeTruncOperator::Operation<TA, TR>(element);
+		return DateTrunc::DecadeOperator::Operation<TA, TR>(element);
 	case DatePartSpecifier::YEAR:
-		return YearTruncOperator::Operation<TA, TR>(element);
+		return DateTrunc::YearOperator::Operation<TA, TR>(element);
 	case DatePartSpecifier::QUARTER:
-		return QuarterTruncOperator::Operation<TA, TR>(element);
+		return DateTrunc::QuarterOperator::Operation<TA, TR>(element);
 	case DatePartSpecifier::MONTH:
-		return MonthTruncOperator::Operation<TA, TR>(element);
+		return DateTrunc::MonthOperator::Operation<TA, TR>(element);
 	case DatePartSpecifier::WEEK:
-		return WeekTruncOperator::Operation<TA, TR>(element);
+		return DateTrunc::WeekOperator::Operation<TA, TR>(element);
 	case DatePartSpecifier::DAY:
-		return DayTruncOperator::Operation<TA, TR>(element);
+	case DatePartSpecifier::DOW:
+	case DatePartSpecifier::ISODOW:
+	case DatePartSpecifier::DOY:
+		return DateTrunc::DayOperator::Operation<TA, TR>(element);
 	case DatePartSpecifier::HOUR:
-		return HourTruncOperator::Operation<TA, TR>(element);
+		return DateTrunc::HourOperator::Operation<TA, TR>(element);
 	case DatePartSpecifier::MINUTE:
-		return MinuteTruncOperator::Operation<TA, TR>(element);
+		return DateTrunc::MinuteOperator::Operation<TA, TR>(element);
 	case DatePartSpecifier::SECOND:
-		return SecondsTruncOperator::Operation<TA, TR>(element);
+	case DatePartSpecifier::EPOCH:
+		return DateTrunc::SecondOperator::Operation<TA, TR>(element);
 	case DatePartSpecifier::MILLISECONDS:
-		return MilliSecondsTruncOperator::Operation<TA, TR>(element);
+		return DateTrunc::MillisecondOperator::Operation<TA, TR>(element);
 	case DatePartSpecifier::MICROSECONDS:
 		// Since microseconds are not stored truncating to microseconds does the same as to milliseconds.
-		return MilliSecondsTruncOperator::Operation<TA, TR>(element);
+		return DateTrunc::MillisecondOperator::Operation<TA, TR>(element);
 	default:
-		throw NotImplementedException("Specifier type not implemented");
+		throw NotImplementedException("Specifier type not implemented for DATETRUNC");
 	}
 }
 
-struct DateTruncOperator {
+struct DateTruncBinaryOperator {
 	template <class TA, class TB, class TR>
 	static inline TR Operation(TA specifier, TB date) {
 		return TruncateElement<TB, TR>(GetDatePartSpecifier(specifier.GetString()), date);
 	}
 };
 
+template <typename TA, typename TR>
+static void DateTruncUnaryExecutor(DatePartSpecifier type, Vector &left, Vector &result, idx_t count) {
+	switch (type) {
+	case DatePartSpecifier::MILLENNIUM:
+		UnaryExecutor::Execute<TA, TR, DateTrunc::MillenniumOperator>(left, result, count);
```

Comment:
`MILLENNIUM, CENTURY, DECADE, YEAR, DAY, HOUR, MILLISECONDS` appear to be uncovered in the constant case (the `date_trunc` test case only handles these cases with non-constant specifiers). Could you add some tests for those?

Reply:
Milliseconds was actually incorrect, and microseconds was missing.